### PR TITLE
Support for Private Repositories for spk infra template generation

### LINKED ISSUE #740: As a developer, I want to use `spk infra generate` with private remote directories
**As a:**
Developer

**I want:**
`spk infra generate` to generate Terraform scripts for all the infrastructure for a specific deployment from a private source.

**So that:**
I can manage terraform templates in my private remote repo.

**Describe the solution you'd like:**
`spk infra scaffold` should take into account private repo credentials from a PAT provided through `spk.config`. `spk infra generate` should be able to use the provided information and fetch the latest version of a template with the provided PAT token.

**Acceptance Criteria:**
The CLI command should include the following functionality:
- [x] `spk infra scaffold` checks spk.config.yaml for the `access_token` of `azure_devops` and also check for a `infra_repository`
- [x] If the token is not provided assume a public repo
- [x] If `infra_repository` is not provided or an arg is passed as a `source` for `spk infra scaffold` override the `source` 

**Does this require updates to documentation?:**
Yes.

### PR DESCRIPTION
**Pull request by NathanielRose**
- Support for PAT tokens with AzDO in `spk infra scaffold`
- Closes https://github.com/microsoft/bedrock/issues/740

### CODE REVIEW

File: src/commands/mocks/spk-config.yaml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 # Azure DevOps Configuration
 azure_devops: # Should this live under the `spk` configuration?
-  access_token: "hpe3a9oiswgcodtfdpzfiek3saxbrh5if1fp673xihgc5ap467a" # This is your Personal Access Token with permission to modify and access this private repo. Leave this empty if project is public
+  access_token: "vwri3lobf5gxz7kdwncfk3tqsxupotqf6nx2ihkrdczo7t4yru3q" # This is your Personal Access Token with permission to modify and access this private repo. Leave this empty if project is public
```
Comment (jmspring) [inline src/commands/mocks/spk-config.yaml]: I assume this isn't a real token?

**Review by jmspring — APPROVED** [review]
One question, otherwise lgtm

File: src/commands/mocks/spk-config.yaml (hunk 1 above)
Comment (NathanielRose) [reply src/commands/mocks/spk-config.yaml]: Redacted ;)

File: spk-config.yaml [hunk 2]
```diff
@@ -1,8 +1,9 @@
 # Azure DevOps Configuration
 azure_devops:
-  access_token: "hpe3a9oiswgcodtfdpzfiek3saxbrh5if1fp673xihgc5ap467a" # This is your Personal Access Token with permission to modify and access this private repo. Leave this empty if project is public
+  access_token: "<my_token>" # This is your Personal Access Token with permission to modify and access this private repo. Leave this empty if project is public
   hld_repository: "dev.azure.com/bhnook/fabrikam/_git/hld" # Repository URL for your Bedrock HLDs
   manifest_repository: "dev.azure.com/bhnook/fabrikam/_git/materialized" # Repository URL that is configured for flux. This holds the kubernetes manifests that is generated by fabrikate.
+  infra_repository: "dev.azure.com/naros32/bedrock_testing_phase2/_git/spk-testing-private-azdo" # Repository URL that contains your terraform templates to be sed for scaffolding and generating infrastructure deployment templates.
   org: "epicstuff" # Your AzDo Org
```
Comment (yradsmikham) [inline spk-config.yaml:7]: Will there ever be a situation where more than one PATs are used? (e.g. different organization, different repos)

File: docs/cloud-infra-management.md [hunk 3]
```diff
@@ -155,3 +155,16 @@ It will do the following (**In Progress**):
...
+Spk currently supports the use of Personal Access Tokens to authenticate with
+private infrastructure repositories hosted in Azure DevOps. To configure spk to
+build scaffolded definitions using a private AzDO repo, do one of the following:
+
+- **Using `.spk-config`** - Pass in your PAT through an .env when you initialize
+  spk. Be sure that the `access_token` and `infra_repository` is set and for
+  every scaffold specify your `--version` and `--template`
+- **Using arguments** - Pass in your formatted source url for your private AzDO
+  repo with the PAT and arbitrary username specified. Example
+  `spk infra scaffold --name discovery-service --source https://sk:{my_PAT_Token}@dev.azure.com/microsoft/spk/_git/infra_repo --version v0.0.1 --template cluster/environments/azure-single-keyvault`
```
Comment (yradsmikham) [inline docs/cloud-infra-management.md]: why is the source `https://sk:`?

File: src/commands/infra/scaffold.ts [hunk 4]
```diff
@@ -32,12 +33,36 @@ export const scaffoldCommandDecorator = (command: commander.Command): void => {
...
+          } else {
+            logger.error(
+              "You must specify each of the variables 'name', 'source', 'version', 'template' in order to scaffold out a deployment."
+            );
+          }
+        } else {
+          if (!opts.source) {
+            // Construct the source based on the the passed configurations of spk-config.yaml
+            opts.source =
+              "https://spk:" +
```
Comment (yradsmikham) [inline src/commands/infra/scaffold.ts:58]: and then down here it's `https://spk:`?

Why is `spk` in the source url?

**Review by yradsmikham — CHANGES_REQUESTED** [review]

File: spk-config.yaml (hunk 2 above)
Comment (NathanielRose) [reply spk-config.yaml:7]: There may be, but we can recommend users to consolidate their templates into one private repo for the first iteration. What do you think?

File: src/commands/infra/scaffold.ts (hunk 4 above)
Comment (NathanielRose) [reply src/commands/infra/scaffold.ts:58]: When using PATs, you need to specify an arbitrary username associated with the token. I am using spk.

**Review by yradsmikham — APPROVED** [review]

**Pull request merged**